backend/danswer/connectors/google_drive/connector.py

```diff
@@ -154,125 +189,167 @@ def load_credentials(self, credentials: dict[str, Any]) -> dict[str, str] | None
         )
         return new_creds_dict
 
-    def _get_all_user_emails(self) -> list[str]:
+    def _update_traversed_parent_ids(self, folder_id: str) -> None:
+        self._retrieved_ids.add(folder_id)
+
+    def _get_all_user_emails(self, admins_only: bool) -> list[str]:
         admin_service = get_admin_service(
             creds=self.creds,
             user_email=self.primary_admin_email,
         )
+        query = "isAdmin=true" if admins_only else "isAdmin=false"
         emails = []
         for user in execute_paginated_retrieval(
             retrieval_function=admin_service.users().list,
             list_key="users",
             fields=USER_FIELDS,
             domain=self.google_domain,
+            query=query,
         ):
             if email := user.get("primaryEmail"):
                 emails.append(email)
         return emails
 
-    def _fetch_drive_items(
+    def _get_all_drive_ids(self) -> set[str]:
+        primary_drive_service = get_drive_service(
+            creds=self.creds,
+            user_email=self.primary_admin_email,
+        )
+        all_drive_ids = set()
+        for drive in execute_paginated_retrieval(
+            retrieval_function=primary_drive_service.drives().list,
+            list_key="drives",
+            useDomainAdminAccess=True,
+            fields="drives(id)",
+        ):
+            all_drive_ids.add(drive["id"])
+        return all_drive_ids
+
+    def _initialize_all_class_variables(self) -> None:
+        # Get all user emails
+        # Get admins first becuase they are more likely to have access to the most files
+        user_emails = [self.primary_admin_email]
+        for admins_only in [True, False]:
+            for email in self._get_all_user_emails(admins_only=admins_only):
+                if email not in user_emails:
+                    user_emails.append(email)
+        self._all_org_emails = user_emails
```

Comment:
do not like this, let's avoid using these stateful instance variables as much as possible

Reply:
let's pass this around to not create this implicit dependency